compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -2409,7 +2410,8 @@ pub struct Constant<'tcx> {
     pub literal: ConstantKind<'tcx>,
 }
 
-#[derive(Clone, Copy, PartialEq, PartialOrd, TyEncodable, TyDecodable, Hash, HashStable, Debug)]
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, TyEncodable, TyDecodable, Hash, HashStable, Debug)]
+#[derive(Lift)]
 pub enum ConstantKind<'tcx> {
```

Comment:
Can you add a doc comment here (and at `ty::ConstKind`) saying that these are just reflecting the variants of `mir/ty::Const`, similar to `Ty` and `TyKind`? Maybe I'm dense but it took me a while to realize that.^^